nixos/modules/services/monitoring/vmagent.nix

```diff
@@ -33,7 +93,7 @@ in {
       };
       basicAuthPasswordFile = lib.mkOption {
         default = null;
-        type = lib.types.nullOr lib.types.str;
+        type = lib.types.either lib.types.str lib.types.path;
```

Comment:
doesnt that mean that i can not manually set `basicAuthPasswordFile = null` if i want to explicitly set the default value again? I think the default should be included into the valid types 
(havent acutally tested as i am currently struggling with my system setup due to an regession)

Reply:
I still think that types.path for secrets is a huge footgun. As soon as somebody does `${foo}` rather than `toString foo` in a subsequent change somewhere, the file will be written into the store __without a warning__.

Reply:
Oops, I didn’t realize that.